### DISCUSSION TITLE
Could not find Function with id: XXX

### DISCUSSION THREAD

**Discussion by lukeclifton** [Shopify/function-examples#249] [Feedback: Cart Transform API]
Thanks for all the amazing work on Shopify Functions.

I am trying to build a cart transform function in Typescript.

I have deployed my cart transform function extension and its now showing as published.

When i try and run the `cartTransformCreate` migration as shown here: https://shopify.dev/docs/apps/selling-strategies/bundles/add-a-customized-bundle#step-4-create-the-carttransform-object-using-graphql

I am getting the following error:

```
"message": "Could not find Function with id: xxxxxxx"
```

I have ensured that the function id is correct and that my app has the `write_products` scope.

What else could wrong? Any ideas? Thanks!

**Comment by ignacio-chiazzo** [comment]
👋 Could you share your request ID?

**Reply by lukeclifton** [reply]
Thanks @ignacio-chiazzo Is there a way to see the request id when using the **Shopify GraphiQL App**?

**Reply by ignacio-chiazzo** [reply]
Sorry, could you share the shop ID and app ID? 

Did you use the same app that deployed the function to make the request via GraphiQL?

**Reply by lukeclifton** [reply]
Sure. 

Shop ID: `gid://shopify/Shop/75388027174`
App ID: `gid://shopify/App/39351517185`

No i deployed the function via the above app ID and made the `cartTransformCreate` request via the Shopify GraphiQL App. 

Is this the reason perhaps?

**Reply by ignacio-chiazzo** [reply]
> No i deployed the function via the above app ID and made the cartTransformCreate request via the Shopify GraphiQL App.

Oh I see. This won't work since the appID used there is `Shopify GraphiQL App`, you would need to execute the query using the app you just created (I assumed the one you linked above).

**Reply by ignacio-chiazzo** [reply]
We have updated Shopify GraphiQL App to create `cartTransforms`. 

If that does not work, let us know and try using your own App to create the `cartTransform` via an API call.

**Reply by lukeclifton** [reply]
> Oh I see. This won't work since the appID used there is `Shopify GraphiQL App`, you would need to execute the query using the app you just created (I assumed the one you linked above).

Thanks @ignacio-chiazzo for your help. This solved the issue. Much appreciated.

**Comment by ignacio-chiazzo** [comment]
Usually, it indicates that the app wasn’t actually installed, or missing access scope.
